### DISCUSSION TITLE
Clang++ LDFLAGS, CXXFLAGS, CFLAGS

### DISCUSSION THREAD

**Discussion by uNetworkingAB** [microsoft/vcpkg#27656] [Ideas]
After installing a package, you need to pass -L and -I flags to Clang/Clang++ for it to know about the package. It would be very minimal and easy, if vcpkg would either set LDFLAGS, CXXFLAGS and CFLAGS directly, or if you prefer some kind of VCPKG_LDFLAGS, VCPKG_CXXFLAGS, VCPKG_CFLAGS.

Note that vcpkg would only set (capital L, capital I) -L and -I (library path, include path) but leave the "link to" flag (lower case L) out of it.

Then we who use raw Clang can just pass those environment variables to the clang command (which we already do) and have everything work just as easy as if we were on a Linux dist.

**Comment by uNetworkingAB** [comment]
Here is an example how I do it currently:

vcpkg install libuv:x64-windows
$Env:CFLAGS='-I C:/vcpkg/packages/libuv_x64-windows/include' **<- not very portable**
$Env:LDFLAGS='-L C:/vcpkg/packages/libuv_x64-windows/lib' **<- not very portable**

(now I use clang)

**Reply by Neumann-A** [reply]
Also you shouldnt use the packages folder. That is just vcpkgs staging dir  use the installed dir instead.

**Comment by Neumann-A** [comment]
Use pkg-config for whatever you are consuming.
